metrics/ci_integrations/test/destination/firestore/config/validation_delegate/firestore_destination_validation_delegate_test.dart

```diff
@@ -0,0 +1,360 @@
+    final firebaseAuth = FirebaseAuthMock();
+
+    final delegate = FirestoreDestinationValidationDelegate(authFactory);
+
+    PostExpectation<FirebaseAuth> whenCreateFirebaseAuth() {
+      return when(authFactory.create(apiKey));
+    }
+
+    PostExpectation<Future<User>> whenSignIn() {
+      whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+      return when(firebaseAuth.signIn(email, password));
+    }
+
+    tearDown(() {
+      reset(authFactory);
+      reset(firebaseAuth);
+    });
+
+    test(
+      "throws an ArgumentError if the given auth factory is null",
+      () {
+        expect(
+          () => FirestoreDestinationValidationDelegate(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "creates an instance with the given parameters",
+      () {
+        final validationDelegate = FirestoreDestinationValidationDelegate(
+          authFactory,
+        );
+
+        expect(validationDelegate.authFactory, equals(authFactory));
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() creates a firebase auth with the firebase auth factory",
+      () {
+        whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+
+        delegate.validatePublicApiKey(apiKey);
+
+        verify(authFactory.create(apiKey)).called(once);
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() signs in with the created firebase auth",
+      () {
+        whenCreateFirebaseAuth().thenReturn(firebaseAuth);
+
+        delegate.validatePublicApiKey(apiKey);
+
+        verify(firebaseAuth.signIn(any, any)).called(once);
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a failure field validation result, if the auth throws a firebase auth exception with the 'invalid api key' code",
+      () async {
+        whenSignIn().thenAnswer(
+          (_) => Future.error(invalidApiKeyAuthException),
+        );
+
+        final result = await delegate.validatePublicApiKey(apiKey);
+
+        expect(result.isFailure, isTrue);
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a result with the 'api key invalid' additional context, if the auth throws a firebase auth exception with the 'invalid api key' code",
+      () async {
+        whenSignIn().thenAnswer(
+          (_) => Future.error(invalidApiKeyAuthException),
+        );
+
+        final result = await delegate.validatePublicApiKey(apiKey);
+
+        expect(
+          result.additionalContext,
+          equals(FirestoreStrings.apiKeyInvalid),
+        );
+      },
+    );
+
+    test(
+      ".validatePublicApiKey() returns a successful field validation result, if the auth throws a firebase auth exception with the exception code that does not indicate the invalid api key",
+      () async {
+        final exceptionCodes = FirebaseAuthExceptionCode.values.where(
+          (code) => code != FirebaseAuthExceptionCode.invalidApiKey,
+        );
+
+        for (final exceptionCode in exceptionCodes) {
```

Comment:
I believe there is no need for testing each error code that does not indicate that the `API key` is invalid. I think we can test this case just using a single error code that does not indicate that the `API key` is invalid.